dune-build.sh

```diff
@@ -1,3 +1,5 @@
+#!/bin/sh
+
```

Comment:
No set -eu here?

Reply:
`$EXE` is unbound, because it is commented-out in [dune-defs.sh](https://github.com/herd/herdtools7/blob/0607078b1bddcaa9e74dcae5168fb816652e102e/dune-defs.sh#L21). The same unbound `$EXE` exists in [dune-uninstall.sh](https://github.com/herd/herdtools7/blob/0607078b1bddcaa9e74dcae5168fb816652e102e/dune-uninstall.sh#L28), so I was going to fix those together in a separate PR.

Reply:
Fine, I merge.